### ISSUE TITLE
Incorrect use of `lookup` on terraform objects instead of maps

### ISSUE THREAD

**Issue by slai11** [terraform-google-modules/terraform-google-lb-http#436]
### TL;DR

In https://github.com/terraform-google-modules/terraform-google-lb-http/blob/master/modules/serverless_negs/main.tf#L193, `lookup(each.value, "protocol", "HTTP")` does not default to `"HTTP"` if `protocol` is undefined in `var.backends`. 

This extends to other instances where `lookup` is used. This is because [lookup expects a map](https://developer.hashicorp.com/terraform/language/functions/lookup).

### Expected behavior

Expected `protocol` to be set to `"HTTP"` if not defined in `var.backend`.

### Observed behavior

`protocol` is set to `null` since `each.value` in `lookup(each.value, "protocol", "HTTP")` is an object. 

### Terraform Configuration

```hcl
Short reproducible snippet


terraform {
  required_version = ">= 1.8"
}

output "nullout" {
  value = null
}

output "object_protocol" {
  value = lookup(var.object_backend, "protocol", "override")
}

output "object_id" {
  value = lookup(var.object_backend, "id", "override")
}

output "map_protocol" {
  value = lookup(var.map_backend, "protocol", "override")
}

output "map_id" {
  value = lookup(var.map_backend, "id", "override")
}

variable "object_backend" {
  type = object({
    id       = string
    protocol = optional(string)
    port     = optional(string, null)
  })

  default = { id : "hi" }
}

variable "map_backend" {
  type = map(string)

  default = { id : "hi" }
}

```

Gives the following with no `object_protocol` output since it is a null.

```
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

map_id = "hi"
map_protocol = "override"
object_id = "hi"
```
```


### Terraform Version

```sh
➜ terraform version
Terraform v1.8.2
on darwin_arm64

Your version of Terraform is out of date! The latest version
is 1.9.1. You can update by downloading from https://www.terraform.io/downloads.html
```

### Additional information

Perhaps we could define the defaults in `var.backends` and drop the use of lookups? Happy to help if the team is at capacity.
